src/components/ConnectionStatus.svelte

```diff
@@ -1,18 +1,34 @@
 <script lang="ts">
-  export let connected = false;
+  import { Network, allNetworks, networkToInfo, selectedNetwork } from '$lib/stores/networksStore';
+  import networkSwitchIcon from '$lib/assets/switch-network.svg';
+  import { isLoggedIn } from '$lib/stores';
+  import { pageContent } from '$lib/stores/pageContentStore';
+
+  let networkName:string | undefined;
+  let networkEndpointURL:string | undefined;
+  selectedNetwork.subscribe(value => {
+    networkName = networkToInfo[value].name;
+    networkEndpointURL = networkToInfo[value].endpoint?.toString()?.replace(/\/$/, '');
+    });
+
+function switchNetwork() {
+  console.log('switchNetwork');
+  pageContent.reconnect();
+}
+
 </script>
 
-<div class="flex gap-6 items-center">
+<div class="flex gap-6 items-center" class:hidden={!$isLoggedIn}>
```

Comment:
When you define an element to be "hidden", it still renders the code even though it doesn't show it. I would recommend using the if statement that is commented out below as to only render this component if we need to.